udata/forms/fields.py

```diff
@@ -200,10 +198,29 @@ def __getattr__(self, name):
 
 class FormField(FieldHelper, fields.FormField):
     def __init__(self, form_class, *args, **kwargs):
+        self.has_data = False
         super(FormField, self).__init__(FormWrapper(form_class),
                                         *args, **kwargs)
 
+    def process(self, formdata, data=unset_value):
+        self._formdata = formdata
+        return super(FormField, self).process(formdata, data=data)
+
+    def validate(self, form, extra_validators=tuple()):
```

Comment:
I kept the upstream signature. I think there is some iteration in `super().validate()`